src/circuit/auto-relay.js

```diff
@@ -263,6 +261,18 @@ class AutoRelay {
       log.error(err)
     }
   }
+
+  /**
+   * @param {PeerId} peerId
+   */
+  async _tryToListenOnRelay (peerId) {
+    try {
+      const connection = await this._libp2p.dial(peerId)
+      await this._addListenRelay(connection, peerId.toB58String())
+    } catch (err) {
+      log.error(`could not connect and listen on known hop relay ${peerId.toB58String()}`)
```

Comment:
added an onError 👍🏼  It would be super cool to have a general error handling approach, as this type of handling is common. In this case, the auto relay lifecycle does not care about the errors as it can try other peers.